### PR TITLE
Fix 5577

### PR DESCRIPTION
**Pull request by jonesh66**
## Proposed changes

This change is intended to let the runner send the onRunnerEnd event while it is shutting down. This will be fixing https://github.com/webdriverio/webdriverio/issues/5577

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Review by christian-bromann — CHANGES_REQUESTED** [review]
@jonesh66 thanks for providing this fix. We need to ensure we have tests for this otherwise changes might cause regressions. Can you add tests for the code changes in this PR?

File: packages/wdio-reporter/src/index.js [hunk 1]
```diff
@@ -193,7 +200,8 @@ export default class WDIOReporter extends EventEmitter {
     /**
      * function to write to reporters output stream
      */
-    write(content) {
+    write (content) {
+        if(content) this.contentPresent = true
```
Comment (christian-bromann) [inline packages/wdio-reporter/src/index.js:204]:
```suggestion
        if(content) {
            this.contentPresent = true
        }

```

**Comment by christian-bromann** [comment]
I think we have a rebase issue here? Do you need help resolve it?

**Comment by jonesh66** [comment]
@christian-bromann yes, I was trying to rebase my changes on top of master before pushing them. Github UI says there are conflicts but I can't find them locally. I pulled latest master on my fork. Any suggestions?

**Comment by christian-bromann** [comment]
@jonesh66 your master branch is outdated, files in `@wdio/reporter` have been ported to TypeScript. I think the best idea is to start from a fresh branch and port of the changes to that. Let me know if you need help with that.

**Comment by christian-bromann** [comment]
Let's close this in favor of #6060

**Pull request closed without merging**